packages/react-native-editor/__device-tests__/helpers/utils.js

```diff
@@ -452,6 +449,34 @@ const toggleOrientation = async ( driver ) => {
 	}
 };
 
+const isEditorVisible = async ( driver ) => {
+	const blockLocator = isAndroid()
+		? `//android.widget.EditText[contains(@content-desc, "Post title")]`
+		: `(//XCUIElementTypeScrollView/XCUIElementTypeOther/XCUIElementTypeOther[contains(@name, "Post title")])`;
+
+	await waitForVisible( driver, blockLocator );
+};
+
+const waitForVisible = async ( driver, blockLocator, iteration = 0 ) => {
+	const maxIteration = 25;
+	const timeout = 1000;
+
+	if ( iteration >= maxIteration ) {
+		throw new Error(
+			`"${ blockLocator }" is still not visible after ${ iteration } retries!`
+		);
+	} else if ( iteration !== 0 ) {
+		// wait 1 second before trying to locate element again
+		await driver.sleep( timeout );
+	}
+
+	const locator = await driver.elementsByXPath( blockLocator );
```

Comment:
I think we could check if the element is visible before waiting, this way if it's already present we'd save waiting one second.

Reply:
Yup, it shouldn't wait the first time, it only starts waiting after the first time it fails because of the condition `( iteration !== 0 )`.

Reply:
Oh true, I overlooked that condition, sorry for the comment. I'm going to resolve the conversation 🙇 .